docs/user-guide/administration/design-table.md

```diff
@@ -19,220 +19,257 @@ Before proceeding, please review the GreptimeDB [Data Model Documentation](/user
-2. Filtering data, such as using the `city` column to view the average temperature in New York City over the past week. This is similar to the `WHERE` clause in SQL.
-3. Grouping and aggregating data. For instance, if the temperature data includes a `state` label in addition to `city`, you can group the data by `state` and calculate the average temperature for each `state` over the past week. This is similar to the `GROUP BY` clause in SQL.
+### Choosing tag columns
 
-Recommendations for Tag columns:
+Usually you don't need tag columns since ordering data by timestamp is sufficient for most use cases.
+If you need deduplication, or your queries can benefit from the ordering, you can define tag columns.
+Setting tags may also reduce disk space usage as it improves the locality of data.
 
-- Typically strings, avoiding `FLOAT` or `DOUBLE`.
-- The number of Tag columns in a table should be moderate, usually not exceeding 20.
-- Control the number of unique values in Tag columns to prevent high cardinality issues, which can negatively impact write performance and lead to index expansion.
-- Ensure that Tag column values remain relatively stable and do not change frequently. For instance, avoid using dynamic values such as serverless container host names as Tag columns.
 
-### Field Columns
+In order to improve sort and deduplication speed under time-series workloads, GreptimeDB buffers and processes rows by time-series internally.
+So it doesn't need to compare the primary key for each row repeatedly.
+This can be a problem if the tag column has high cardinality:
 
-Field columns generally carry the actual measured values. For example, the temperature value in weather data should be set as a Field column. In monitoring systems, CPU utilization, memory utilization, etc., are typical Field columns.
+1. Performance degradation since the database can't batch rows efficiently.
+2. It may increase memory and CPU usage as the database has to maintain the metadata for each time-series.
+3. Deduplication may be too expensive.
 
-Characteristics of Field columns:
 
-1. Usually numerical types (integers, floating-point numbers), logs, and event messages are generally strings.
-2. Used for calculations and aggregations.
-3. Can change frequently, meaning they can have any cardinality.
+Recommendations:
 
-Recommendations for Field columns:
+- Low cardinality columns that occur in `WHERE`/`GROUP BY`/`ORDER BY` frequently.
+  These columns usually remain constant.
+  For example, `namespace`, `cluster`, or an AWS `region`.
+- No need to set all low cardinality columns as tags since this may impact the performance of ingestion and querying.
+- Typically use short strings for tags, avoiding `FLOAT`, `DOUBLE`, `TIMESTAMP`.
+- Never set high cardinality columns as tags if they change frequently.
+  For example, `trace_id`, `span_id`, `user_id` must not be used as tags.
+  GreptimeDB works well if you set them as fields instead of tags.
```

Comment:
Note: Creating indexes on fields and tags can be done at any time.

Reply:
Some content for reference: https://clickhouse.com/docs/use-cases/observability/schema-design#choosing-a-primary-ordering-key

Reply:
Currently, this is an implementation limitation. Our primary key is a time-series and is similar to a stream in VL.
https://docs.victoriametrics.com/victorialogs/keyconcepts/#how-to-determine-which-fields-must-be-associated-with-log-streams

In the future, we can support flexible primary key, but some optimization is only possible and available when the cardinality is low,

For `trace_id+spanid`, it is actually not a time-series. We may need to consider this in the future.